projects/com.oracle.truffle.llvm.test/src/com/oracle/truffle/llvm/test/interop/LLVMTckTest.java

```diff
@@ -0,0 +1,410 @@
+import com.oracle.truffle.llvm.test.LLVMPaths;
+import com.oracle.truffle.llvm.tools.Clang;
+import com.oracle.truffle.llvm.tools.Clang.ClangOptions;
+import com.oracle.truffle.llvm.tools.Opt;
+import com.oracle.truffle.llvm.tools.Opt.OptOptions;
+import com.oracle.truffle.llvm.tools.Opt.OptOptions.Pass;
+import com.oracle.truffle.tck.TruffleTCK;
+
+public class LLVMTckTest extends TruffleTCK {
+    private static final String FILENAME = "tck";
+
+    @Test
+    public void testVerifyPresence() {
+        PolyglotEngine vm = PolyglotEngine.newBuilder().build();
+        assertTrue("Our language is present", vm.getLanguages().containsKey(LLVMLanguage.LLVM_IR_MIME_TYPE));
+        vm.dispose();
+    }
+
+    @Override
+    protected PolyglotEngine prepareVM(Builder builder) throws Exception {
+        PolyglotEngine engine = builder.build();
+        try {
+            File cFile = new File(LLVMPaths.INTEROP_TESTS, FILENAME + ".c");
+            File bcFile = File.createTempFile(LLVMPaths.INTEROP_TESTS + "/" + "bc_" + FILENAME, ".ll");
+            File bcOptFile = File.createTempFile(LLVMPaths.INTEROP_TESTS + "/" + "bcopt_" + FILENAME, ".ll");
+            Clang.compileToLLVMIR(cFile, bcFile, ClangOptions.builder());
+            Opt.optimizeBitcodeFile(bcFile, bcOptFile, OptOptions.builder().pass(Pass.MEM_TO_REG));
+            engine.eval(Source.newBuilder(bcOptFile).build()).as(Integer.class);
+        } catch (IOException e) {
+            throw new AssertionError(e);
+        }
+        return engine;
+    }
+
+    @Override
+    protected String mimeType() {
+        return LLVMLanguage.LLVM_IR_MIME_TYPE;
+    }
+
+    @Override
+    protected String fourtyTwo() {
+        return "fourtyTwo";
+    }
+
+    @Override
+    protected String identity() {
+        return "identity";
+    }
+
+    @Override
+    protected String plus(Class<?> type1, Class<?> type2) {
+        return "plus";
+    }
+
+    @Override
+    protected String returnsNull() {
+        return "returnsNull";
+    }
+
+    @Override
+    protected String applyNumbers() {
+        return "apply";
+    }
+
+    @Override
+    protected String compoundObject() {
+        return "compoundObject";
+    }
+
+    @Override
+    protected String valuesObject() {
+        return "valuesObject";
+    }
+
+    @Override
+    protected String invalidCode() {
+        // @formatter:off
+        return
+            "f unction main() {\n" +
+            "  retu rn 42;\n" +
+            "}\n";
+        // @formatter:on
+    }
+
+    @Override
+    protected String complexAdd() {
+        return "complexAdd";
+    }
+
+    @Override
+    protected String multiplyCode(String firstName, String secondName) {
+        // @formatter:off
+        /*
+        return
+            "function multiply(" + firstName + ", " + secondName + ") {\n" +
+            "  return " + firstName + " * " + secondName + ";\n" +
+            "}\n";
+        // @formatter:on
+         */
+        return null;
```

Comment:
Because there is some constraint in the eval-code, and therefore this test case does not work right now, and I also did not yet figure out how to fix it.